### ISSUE TITLE
Validation

### ISSUE THREAD

**Issue by meehkal** [langdoc/sjd-gold#5]
We need to validate the files for consistency: a) are the minimal annotations included (are the relevant tiers included, do these tiers include data, do the data in these tiers confirm with our conventions [e.g. tags for pos and morph), b) is the XML according to our schema (tier structure, etc.), c) is there metadata for these files?

In order to validate, we need defined conventions for: a) minimal annotations, b) XML schema, c) metadata.

**Comment by nikopartanen** [comment]
This is very important. The tests could be written in Python so they have maximal audience. I assume the structure would be a series of independent small tests, which would be run together to test various things. This way it would also be less tied into our project structure. 

- Test that file names follow the pattern
- Tier types present
- Tiers present
- Participant names matching to metadata (part before @ in tier name)
- Checking whether 'illegal' characters occur, i.e. LATIN SMALL LETTER O WITH DIAERESIS instead of CYRILLIC SMALL LETTER O WITH DIAERESIS on Komi tier. This may be a bad test though, there may be Finnish and German on tiers marked for Komi…
    - Other variant could be to check whether LATIN SMALL LETTER O WITH DIAERESIS appears adjacent to Cyrillic characters, meaning it is in the same word, and thereby a mistake

**Comment by meehkal** [comment]
Actually, this validation should be done in (private) https://github.com/langdoc/langdoc/sjd/, where all the original files live. Once the structure of these files is fine, they can be easily move here.

**Comment by nikopartanen** [comment]
Yes, this is good point! We can also have the tests somewhere separate so at least most generic ones can be easily used across all our projects, but we can think about this later!
